### DISCUSSION TITLE
Request sent event

### DISCUSSION THREAD

**Discussion by iamoskvin** [sindresorhus/got#2382] [Q&A] [answered]
Is there a way to subscribe on event: "the request is sent"?

**Comment by sindresorhus** [comment, marked as answer]
Yes, you can use the `request` event which fires when the request is sent:

```javascript
const request = got('https://example.com');

request.on('request', (request) => {
	console.log('Request sent:', request.requestUrl);
});

const response = await request;
```

Keep in mind the request object passed to the event is the underlying Node.js request (http.ClientRequest), so you can access things like headers, method, etc. from it.

Alternatively, you could use the `beforeRequest` hook if you need to do someting right before the request is sent.
